akka-http-core/src/main/scala/akka/http/impl/engine/client/OutgoingConnectionBlueprint.scala

```diff
@@ -324,7 +328,13 @@ private[http] object OutgoingConnectionBlueprint {
       val getNextMethod = () ⇒ {
         waitingForMethod = true
         if (isClosed(bypassInput)) completeStage()
-        else pull(bypassInput)
+        responseContexts.pop()
+        if (responseContexts.size() > 0) {
+          val responseContext = responseContexts.getFirst()
+          parser.setContextForNextResponse(responseContext)
+          val output = parser.parseBytes(ByteString.empty)
+          drainParser(output)
+        }
```

Comment:
and then pull here again if we're below the demand

    if (responseContexts.size() == pipelineLimit-1) pull(bypassInput)